plugins/yum/nexus-yum-repository-plugin/src/main/java/org/sonatype/nexus/yum/internal/YumHostedImpl.java

```diff
@@ -107,10 +110,10 @@ public YumHostedImpl(final TaskScheduler nexusScheduler,
 
     this.aliases = Maps.newHashMap();
 
-    this.baseDir = RepositoryUtils.getBaseDir(repository);
-
     this.yumGroupsDefinitionFile = null;
 
+    this.yumStore = checkNotNull(yumStoreFactory).create(repository.getId());
```

Comment:
Not much of value in checkNotNull here, as same effect with same NPE would be thrown if not used at all?